python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -0,0 +1,491 @@
+        self.input_lengths = meta["input_lengths"]
+
+
+class GeoColumn(NumericalColumn):
+    """
+    Parameters
+    ----------
+    data : A GeoArrowBuffers object
+    meta : A GeoPandasMeta object (optional)
+
+    Notes
+    -----
+    The GeoColumn class subclasses `NumericalColumn`. Combined with
+    `_copy_type_metadata`, this assures support for existing cudf algorithms.
+    """
+
+    def __init__(
+        self,
+        data: GeoArrowBuffers,
+        meta: GeoPandasMeta = None,
+        shuffle_order: cudf.Index = None,
+    ):
+        base = cudf.Series(cudf.RangeIndex(0, len(data)))._column.data
+        super().__init__(base, dtype="int64")
+        self._geo = data
+        if meta is not None:
+            self._meta = meta
+        else:
+            self._meta = GeoMeta(data)
+        if shuffle_order is not None:
+            self._data = shuffle_order
+
+    def __iter__(self):
+        self._iter_index = 0
+        return self
+
+    def __next__(self):
+        if self._iter_index >= len(self):
+            raise StopIteration
+        result = self[self._iter_index]
+        self._iter_index = self._iter_index + 1
+        return result
+
+    def to_host(self):
+        result = GeoColumn(self._geo.to_host(), self._meta.copy(), self.data)
+        return result
+
+    class GeoColumnLocIndexer:
+        """
+        Not yet supported.
+        """
+
+        def __init__(self):
+            # Todo: Easy to implement with a join.
+            raise NotImplementedError
+
+    class GeoColumnILocIndexer:
+        """
+        Each row of a GeoSeries is one of the six types: Point, MultiPoint,
+        LineString, MultiLineString, Polygon, or MultiPolygon.
+        """
+
+        def __init__(self, sr):
+            self._sr = sr
+
+        def __getitem__(self, index):
+            if not isinstance(index, slice):
+                mapped_index = int(self._sr.values[index])
+                return self._getitem_int(mapped_index)
+            else:
+                raise NotImplementedError
+                # This slice functionality is not ready yet
+                # return self._getitem_slice(index)
+
+        def _getitem_int(self, index):
+            type_map = {
+                "p": gpuPoint,
+                "mp": gpuMultiPoint,
+                "l": gpuLineString,
+                "ml": gpuMultiLineString,
+                "poly": gpuPolygon,
+                "mpoly": gpuMultiPolygon,
+            }
+            return type_map[self._sr._types[index]](self._sr, index)
+
+    def __getitem__(self, item):
+        """
+        Returns gpuGeometry objects for each of the rows specified by index.
+        """
+        if not isinstance(item, numbers.Integral):
+            raise NotImplementedError
+        return self.iloc[item]
+
+    @property
+    def loc(self):
+        """
+        Not currently supported.
+        """
+        return self.GeoColumnLocIndexer(self)
```

Comment:
If you do choose to keep the nested class, `self.__class__.GeoColumnLocindexer` would be more accurate here, as the type itself isn't bound to this instance (`self`) in any way.